addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationEnvironment.java

```diff
@@ -167,13 +167,29 @@ public String replaceVars(Object value) {
     }
 
     public Map<String, String> replaceMapVars(Map<String, String> map) {
-        Map<String, String> map2 = new HashMap<>();
+        Map<String, String> map2 = new LinkedHashMap<>();
```

Comment:
since we check for `instanceof LinkedHashMap` in other places, we need to retain the type here